### PR TITLE
Remove multiple objects functionality from generate repo

### PR DESCRIPTION
**Pull request by wurzka**

### CODE REVIEW

**Comment by wurzka** [comment]
Thanks for fixing the typo @larshp ☺️

File: src/z_generate_repo.prog.abap [hunk 1]
```diff
@@ -306,139 +305,120 @@ CLASS lcl_generator IMPLEMENTATION.
...
+      ELSE.
+        readme = VALUE rswsourcet(
+            ( |# { object-object_type } File Format| )
+            ( `` )
+            ( `File | Cardinality | Definition | Schema | Example` )
+            ( `:--- | :---  | :--- | :--- | :---` )
+            ( |`<name>.{ object_type_folder_name }.json` \| 1 \| [`{ interfacename }.intf.abap`](./type/{ interfacename }.intf.abap) \| [`{ to_lower( objecttype ) }-v{ format_version }.json`](./{ to_lower( objecttype ) }-v{ format_version }.json)| &&
+              | \| | )
+            ( `` )
+        ) ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT.
```
Comment (huber-nicolas) [inline src/z_generate_repo.prog.abap:417]: Just a minor.
As far as I see, these lines are nearly the same. Maybe we can do it like this:
`data example_part type string.
IF object-example  IS NOT INITIAL.
        data(examplename)= replace_names_in_string( content_as_string = to_lower( object-example ) replacing_table_string = replacing_table_string ).
   example_part = | \| [`{ examplename }.{ object_type_folder_name }.json`](./examples/{ examplename }.{ object_type_folder_name }.json)| .
endif.
DATA(readme) = VALUE rswsourcet(
            ( |# { object-object_type } File Format| )
            ( `` )
            ( `File | Cardinality | Definition | Schema | Example` )
            ( `:--- | :---  | :--- | :--- | :---` )
            ( |`<name>.{ object_type_folder_name }.json` \| 1 \| [`{ interfacename }.intf.abap`](./type/{ interfacename }.intf.abap) \| [`{ to_lower( objecttype ) }-v{ format_version }.json`](./{ to_lower( objecttype ) }-v{ format_version }.json)| &&
               example_part  )
            ( `` )
        ) ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT ##NO_TEXT. `

Comment (wurzka) [reply src/z_generate_repo.prog.abap:417]: Done

**Comment by wurzka** [comment]
I had to reorder some method calls as I run in null pointer exceptions when no example name had been inserted. Therefore, I created a unit test for repo folder INTF without example file

File: src/z_generate_repo.prog.abap [hunk 2]
```diff
@@ -306,139 +303,114 @@ CLASS lcl_generator IMPLEMENTATION.
...
-          DATA(file_handler) = aff_factory->get_object_file_handler( ). " for testing purposes
-        ELSE.
-          file_handler = cl_aff_factory=>get_object_file_handler( ).
-        ENDIF.
-        DATA(example_files) = file_handler->serialize_objects( objects = VALUE #( ( example_main_object ) ) log = aff_framework_log ).
+    IF aff_factory IS NOT INITIAL.
+      DATA(file_handler) = aff_factory->get_object_file_handler( ). " for testing purposes
+    ELSE.
+      file_handler = cl_aff_factory=>get_object_file_handler( ).
+    ENDIF.
```
[the remark this answers is not in the document]
Comment (wurzka) [reply src/z_generate_repo.prog.abap:312]: In my opinion, you can even skip the factory and inject the file handler.

Anyhow, I would move it to a separate pull request. I found similar cases und would like to fix them together.

File: src/z_generate_repo.prog.abap [hunk 3]
```diff
@@ -1567,6 +1540,26 @@ CLASS ltc_generator IMPLEMENTATION.
     assert_logs_and_file_handler( ).
   ENDMETHOD.
 
+    METHOD repo_zip_intf_no_ex.
```
Comment (schneidermic0) [inline src/z_generate_repo.prog.abap:1543]: The indentation seems to be wrong

```suggestion
  METHOD repo_zip_intf_no_ex.
```

**Review by schneidermic0 — CHANGES_REQUESTED** [review]

**Review by schneidermic0 — APPROVED** [review]
Looks good to me, now. Thanks, @wurzka

**Review by huber-nicolas — APPROVED** [review]

**Pull request merged**